From 23671dd1634748181b2d046246ac1c1114bd7b35 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:25:51 -0400
Subject: [tools] perf/build: Remove the volatile-register-var feature check

Message-id: <1408462094-14194-183-git-send-email-jolsa@redhat.com>
Patchwork-id: 88168
O-Subject: [PATCH RHEL7.1 BZ1131394 182/325] tools/perf/build: Remove the volatile-register-var feature check
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 01287e2cb7ad17b3d77751888d458a6b6a2bba15
Author: Ingo Molnar <mingo@kernel.org>
Date: Thu Oct 10 08:58:57 2013 +0200

description
===========
Namhyung Kim noticed that the volatile-register-var feature check
is superfluous:

 > The gcc manpage says this warning is enabled by -Wall, and we add -Wall
 > to CFLAGS before doing feature checks.  So all gcc versions that support
 > -Wvolatile-register-var enables it by default without this check and
 > older gcc versions will always fail the feature check.

Remove it - this will further speed up feature checks.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/config/Makefile b/tools/perf/config/Makefile
index d207922..c516d6b 100644
--- a/tools/perf/config/Makefile
+++ b/tools/perf/config/Makefile
@@ -132,8 +132,7 @@ CORE_FEATURE_TESTS =   \
  libunwind   \
  on-exit    \
  stackprotector   \
- stackprotector-all  \
- volatile-register-var
+ stackprotector-all
 
 #
 # So here we detect whether test-all was rebuilt, to be able
@@ -203,10 +202,6 @@ ifeq ($(feature-stackprotector), 1)
   CFLAGS += -Wstack-protector
 endif
 
-ifeq ($(feature-volatile-register-var), 1)
-  CFLAGS += -Wvolatile-register-var
-endif
-
 ifeq ($(DEBUG),0)
   ifeq ($(feature-fortify-source), 1)
     CFLAGS += -D_FORTIFY_SOURCE=2
diff --git a/tools/perf/config/feature-checks/Makefile b/tools/perf/config/feature-checks/Makefile
index cf33596..2eb8346 100644
--- a/tools/perf/config/feature-checks/Makefile
+++ b/tools/perf/config/feature-checks/Makefile
@@ -25,8 +25,7 @@ FILES=     \
  test-libunwind   \
  test-on-exit   \
  test-stackprotector-all  \
- test-stackprotector  \
- test-volatile-register-var
+ test-stackprotector
 
 CC := $(CC) -MD
 
@@ -37,7 +36,7 @@ BUILD = $(CC) $(LDFLAGS) -o $(OUTPUT)$@ $@.c
 ###############################
 
 test-all:
- $(BUILD) -Werror -fstack-protector -fstack-protector-all -Wvolatile-register-var -O2 -Werror -D_FORTIFY_SOURCE=2 -ldw -lelf -lnuma -lunwind -lunwind-x86_64 -lelf -laudit -I/usr/include/slang -lslang $(shell pkg-config --libs --cflags gtk+-2.0 2>/dev/null) $(FLAGS_PERL_EMBED) $(FLAGS_PYTHON_EMBED) -DPACKAGE='"perf"' -lbfd -ldl
+ $(BUILD) -Werror -fstack-protector -fstack-protector-all -O2 -Werror -D_FORTIFY_SOURCE=2 -ldw -lelf -lnuma -lunwind -lunwind-x86_64 -lelf -laudit -I/usr/include/slang -lslang $(shell pkg-config --libs --cflags gtk+-2.0 2>/dev/null) $(FLAGS_PERL_EMBED) $(FLAGS_PYTHON_EMBED) -DPACKAGE='"perf"' -lbfd -ldl
 
 test-hello:
  $(BUILD)
@@ -48,9 +47,6 @@ test-stackprotector-all:
 test-stackprotector:
  $(BUILD) -Werror -fstack-protector
 
-test-volatile-register-var:
- $(BUILD) -Werror -Wvolatile-register-var
-
 test-fortify-source:
  $(BUILD) -O2 -Werror -D_FORTIFY_SOURCE=2
 
-- 
1.7.1